pkg/storage/storagefactory/storage_factory.go

```diff
@@ -0,0 +1,195 @@
+package storagefactory
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/distribution/registry/storage/driver/factory"
+
+	"zotregistry.io/zot/errors"
+	"zotregistry.io/zot/pkg/api/config"
+	"zotregistry.io/zot/pkg/extensions/monitoring"
+	"zotregistry.io/zot/pkg/log"
+	"zotregistry.io/zot/pkg/storage"
+	"zotregistry.io/zot/pkg/storage/common"
+	"zotregistry.io/zot/pkg/storage/local"
+	"zotregistry.io/zot/pkg/storage/s3"
+)
+
+func New(config *config.Config, linter common.Lint, metrics monitoring.MetricServer,
```

Comment:
Why not promote this up to pkg/storage/?

Reply:
because it needs to import local/s3 for constructors, but at the same time local/s3 needs common code (which needs storage interface).